baseapp/abci.go

```diff
@@ -330,7 +330,7 @@ func (app *BaseApp) DeliverTx(req abci.RequestDeliverTx) (res abci.ResponseDeliv
 	defer func() {
 		for _, streamingListener := range app.abciListeners {
```

Comment:
This isn't your fault @yihuang but `app.abciListeners` is both read-from and written-to without synchronization, which is a memory model violation. Reads occur throughout baseapp/abci.go and writes occur at baseapp/options.go SetStreamingService.

Reply:
the abci methods are not called concurrently, the `SetStreamingService` is supposed to be called on initialization only. So I believe we don't need to be thread-safe here.